spec/services/create_next_edition_service_spec.rb

```diff
@@ -0,0 +1,118 @@
+
+    it "appends the change note details when there is a change note and a major change" do
+      current_edition = create(:edition,
+                               :published,
+                               number: 2,
+                               published_at: Date.yesterday.noon,
+                               change_note: "note",
+                               update_type: "major")
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_history.first).to match a_hash_including(
+        "note" => "note",
+        "public_timestamp" => Date.yesterday.noon.rfc3339,
+      )
+    end
+
+    it "maintains the change history when the current edition is not a major change" do
+      current_edition = create(:edition,
+                               :published,
+                               number: 2,
+                               published_at: Date.yesterday.noon,
+                               change_note: "note",
+                               update_type: "minor")
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_history).to eq([])
+    end
+
+    it "maintains the change history when current edition lacks a change note" do
+      current_edition = create(:edition,
+                               :published,
+                               number: 2,
+                               published_at: Date.yesterday.noon,
+                               change_note: "")
+
+      next_edition = described_class.call(current_edition: current_edition,
+                                          user: user)
+
+      expect(next_edition.change_history).to eq([])
+    end
+
+    context "when the current edition is the first edition" do
+      it "doesn't append the change note details" do
+        current_edition = create(:edition,
+                                 :published,
+                                 published_at: Date.yesterday.noon,
+                                 change_note: "note")
+
+        next_edition = described_class.call(current_edition: current_edition,
+                                            user: user)
+
+        expect(next_edition.change_history).to eq([])
+      end
+    end
+
+    context "when given a discarded edition" do
+      it "updates the state of the discarded edition" do
+        live_edition = create(:edition, :published)
+        discarded_edition = create(:edition,
+                                   state: "discarded",
+                                   current: false,
+                                   document: live_edition.document)
+
+        next_edition = described_class.call(current_edition: live_edition,
+                                            user: user, discarded_edition: discarded_edition)
+
+        expect(next_edition.state).to eq "draft"
```

Comment:
I think this test would be clearer if it didn't assign to next_edition.

Eg:

```
  described_class.call(current_edition: live_edition,
                       user: user, 
                       discarded_edition: discarded_edition)
  expect(discarded_edition).to be_draft
```

Reply:
@kevindew Thanks, yes - that does look clearer and cleaner.  Will fix.